src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -97,6 +101,15 @@ public class GwResource {
     private final Counter receivedActions;
     private final Counter forwardedActions;
 
+    List<SourceEnvironment> sourceEnvironments = new ArrayList<>();
+
+    Map<String, Map<String, List<String>>> mapBaet = new HashMap<>();
```

Comment:
From a maintenance perspective, this could make things a bit clearer:

```
Set<CachedEventType> cachedEventTypes = new HashSet<>();
```

with

```
public class CachedEventType {
  String bundle;
  String application;
  String eventType;
  // constructor, equals and hashCode impls based on all fields above
}
```

and then

```
if (!cachedEventTypes.contains(new CachedEventType(ra.getBundle(), ra.getApplication(), ra.getEventType())) {
  // reject event
}
```